tes-adapter/src/main/java/com/epam/pipeline/tesadapter/service/CloudPipelineAPIClient.java

```diff
@@ -11,73 +11,92 @@
 import com.epam.pipeline.entity.pipeline.run.PipelineStart;
 import com.epam.pipeline.entity.region.AbstractCloudRegion;
 import com.epam.pipeline.rest.PagedResult;
+import com.epam.pipeline.tesadapter.entity.TesTokenHolder;
 import com.epam.pipeline.utils.QueryUtils;
 import com.epam.pipeline.vo.PagingRunFilterExpressionVO;
 import com.epam.pipeline.vo.PagingRunFilterVO;
 import com.epam.pipeline.vo.RunStatusVO;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
 
 import java.util.List;
 
 @Service
 public class CloudPipelineAPIClient {
-    private CloudPipelineAPI cloudPipelineAPI;
+    private TesTokenHolder tesTokenHolder;
 
-    public CloudPipelineAPIClient(@Value("${cloud.pipeline.host}") String cloudPipelineHostUrl,
-                                  @Value("${cloud.pipeline.token}") String cloudPipelineToken) {
-        this.cloudPipelineAPI =
-                new CloudPipelineApiBuilder(0, 0, cloudPipelineHostUrl, cloudPipelineToken)
-                        .buildClient();
+    @Value("${cloud.pipeline.host}")
+    private String cloudPipelineHostUrl;
+
+    @Autowired
+    public CloudPipelineAPIClient(TesTokenHolder tesTokenHolder) {
+        this.tesTokenHolder = tesTokenHolder;
+    }
+
+    private CloudPipelineAPI buildCloudPipelineAPI(String apiHost, String token) {
```

Comment:
since we have `tesTokenHolder` and `cloudPipelineHostUrl` as a state of `CloudPipelineAPIClient` this method can be without any parameters

Reply:
fixed